### PR TITLE
KAFKA 8311: better handle timeout exception on Stream thread

### PR DESCRIPTION
**Pull request by abbccdda**
The goals for this small diff are:

1. Give user guidance if they want to relax commit timeout threshold
2. Indicate the code path where timeout exception was caught

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by abbccdda** [comment]
Retest this please

File: clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java [hunk 1]
```diff
@@ -1761,7 +1761,8 @@ public OffsetAndMetadata committed(TopicPartition partition, final Duration time
                     Collections.singleton(partition), time.timer(timeout));
             if (offsets == null) {
                 throw new TimeoutException("Timeout of " + timeout.toMillis() + "ms expired before the last " +
-                        "committed offset for partition " + partition + " could be determined");
+                        "committed offset for partition " + partition + " could be determined. User could tune default.api.timeout.ms " +
```
Comment (vvcephei) [inline clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java]: The user is the one reading this message. What do you think of something like "You can tune..."? Or "Try tuning..."?

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java [hunk 2]
```diff
@@ -498,6 +498,9 @@ void commit(final boolean startNewTransaction) {
             }
         } catch (final CommitFailedException | ProducerFencedException error) {
             throw new TaskMigratedException(this, error);
+        } catch (final TimeoutException error) {
+            log.error("Main thread times out while doing commit.");
```
Comment (vvcephei) [inline streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java]: Can you elaborate on what this means? It doesn't appear to contain more information than the error itself. What are we trying to convey here?

**Review by vvcephei — COMMENTED** [review]
Thanks! Our error messages in here leave something to be desired. Thanks for improving them.

File: clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java (hunk 1 above)
Comment (abbccdda) [reply clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java]: Sounds good!

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java (hunk 2 above)
Comment (abbccdda) [reply streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java]: Make sense, I think we should remove this log.

**Comment by abbccdda** [comment]
Retest this please

**Comment by abbccdda** [comment]
Retest this please

**Comment by guozhangwang** [comment]
LGTM. Merged to trunk.

**Pull request merged**